packages/jest-cli/src/TestRunner.js

```diff
@@ -238,10 +247,22 @@ class TestRunner {
         aggregatedResults.success =
           aggregatedResults.numFailedTests === 0 &&
           aggregatedResults.numRuntimeErrorTestSuites === 0;
-        if (reporter.onRunComplete) {
-          reporter.onRunComplete(config, aggregatedResults);
-        }
-        return aggregatedResults;
+
+        return this._hasteMap.matchFiles(/\.snap/).then(res => {
+          aggregatedResults.snapshotsUnlinked = 0;
+          res.forEach(snapFile => {
+            const parts = snapFile.split('__snapshots__/');
+            const testFile = parts[0] + parts[1].slice(0, -5);
```

Comment:
these two lines will most definitely break on windows. Look at `path.basename` and `path.dirname`. With two `path.dirname` calls you can go up two levels. With `path.basename(foo, '.' + SNAPSHOT_EXTENSION)` you should be able to get the expected file name.